### ISSUE TITLE
[Question] ClamAV being used internally with NGINX

### ISSUE THREAD

**Issue by Ryu945** [mko-x/docker-clamav#69]
1)  I was looking through your docker compose file.  I assume I don't need the legacy links tag if I put ClamAV on the same network.  My question involves using it with NextCloud and NGINX.  If I make the ClamAV container internal, will NGINX container provide needed network to update ClamAV if I set ClamAV as depends_on NGINX container?

2)  I assume this is just for debian builds and not CentOS builds:
`#debian: image: mkodockx/docker-clamav`

3)  It says you have proxy support but there is no documentation anywhere that I can find on how to use this proxy support.

4)  For some reason, using ClamAV container with a reverse proxy, letsencrypt, database and nextcloud prevents them from finishing their setup.

edit:

I managed to finally get this working but for some reason, I had to delete my volumes for this to happen.  I am not sure the reason why.   Is there any reason a NGINX, Letsencrypt, NextCloud and Mariadb server that was running can not be torn down and remade with their current volumes plus this ClamAV container on the side?

I also noticed a major problem with this image.   It randomly decides to make a volume even though no where in the docker compose file tells it to make a volume.   This means  that whatever is using the volume for cannot be remounted since a mounted volume was never used to create the server in the first place.  It also means that every time I run this container, a volume gets created.

edit2: 

I see the volume is being created by this.  Shouldn't you have this volume made in the docker compose file instead?

```
# volume provision
VOLUME ["/var/lib/clamav"]
```

**Comment by WhiteBahamut** [comment]
I think it is valid to declare the `VOLUME` within the Dockerfile. You still can specify where to mount it (`docker run -v /my/path/:/var/lib/clamav`).
Can you post your compose file?

**Comment by Ryu945** [comment]
I tried that and it just makes two volumes.   Both the one you specified and the one I have listed in my compose file.   The AV part of my compose file looks like this.  Tell me if you need the entire file.

  ```
av:
    image: mkodockx/docker-clamav:alpine
    container_name: nextcloud-clamav
    networks: 
      - network1
    depends_on:
      - reverse_proxy
    volumes:
      - nextcloud_clamav:/var/lib/clamav
    environment:
      - NETWORK_ACCESS=internal
    restart: unless-stopped
```

**Comment by WhiteBahamut** [comment]
I am not the maintainer here ;) Just try to help out!
Wondering why this happens at all, afaik it _should_ not. How have you definded `nextcloud_clamav`? Is it used by another service as well?
On the other hand **I** would just ditch the `VOLUME` declaration in the Dockerfile and add documentation for the sample compose which directory should be mapped. Once my current multi-arch PR is approved I could issue a new PR to remove the Dockerfile `VOLUME`.

**Comment by Ryu945** [comment]
I have defined nextcloud_clamav like a volume.   The volume declaration being removed from the docker file should fix this problem.  This declaration should be in the docker compose file or in examples of terminal commands.

**Comment by WhiteBahamut** [comment]
I would treat the docker-compose provided in here just a very basic example. Will try to work in the volume issue this week and update the sample compose and readme then.
- [ ] remove `VOLUME` from Dockerfiles
- [ ] update README with persistency instruction
- [ ] update sample compose to reflect volume usage

**Comment by mko-x** [comment]
thanks to @WhiteBahamut closed by  #72

**Comment by WhiteBahamut** [comment]
more PRs might come as I start to use the images more myself ;-)
